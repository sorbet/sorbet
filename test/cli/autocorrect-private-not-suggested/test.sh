tmp="$(mktemp -d)"
cp test/cli/autocorrect-private-not-suggested/*.rb "$tmp"

cwd="$(pwd)"
cd "$tmp" || exit 1

if "$cwd/main/sorbet" --silence-dev-message -a autocorrect-private-not-suggested.rb 2>&1; then
  echo "Expected to fail!"
  exit 1
fi

echo
echo --------------------------------------------------------------------------
echo

# verifies that the file is unchanged when suggestions are applied
# i.e. it does not replace the unknown method with a private one
cat autocorrect-private-not-suggested.rb

rm autocorrect-private-not-suggested.rb
rm "$tmp"