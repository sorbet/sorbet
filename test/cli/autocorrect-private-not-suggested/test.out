autocorrect-private-not-suggested.rb:11: Method `my_private_metho` does not exist on `A` https://srb.help/7003
    11 |A.new.my_private_metho
              ^^^^^^^^^^^^^^^^
  Got `A` originating from:
    autocorrect-private-not-suggested.rb:11:
    11 |A.new.my_private_metho
        ^^^^^

autocorrect-private-not-suggested.rb:7: Method `my_private_metho` does not exist on `A` https://srb.help/7003
     7 |    my_private_metho
            ^^^^^^^^^^^^^^^^
  Autocorrect: Done
    autocorrect-private-not-suggested.rb:7: Replaced with `my_private_method`
     7 |    my_private_metho
            ^^^^^^^^^^^^^^^^
    autocorrect-private-not-suggested.rb:4: Defined here
     4 |  private def my_private_method; end
                  ^^^^^^^^^^^^^^^^^^^^^
Errors: 2

--------------------------------------------------------------------------

# typed: true

class A
  private def my_private_method; end

  def example
    my_private_method
  end
end

A.new.my_private_metho
